### DISCUSSION TITLE
How To install the PMD plugin for JDeveloper 12c(12.2.1.4.0)

### DISCUSSION THREAD

**Discussion by genovalee** [pmd/pmd#4076] [Q&A]
Greeting everyone:
PMD just  plugin for JDeveloper 10 and 11.
How To install the PMD plugin for JDeveloper 12c?

**Comment by adangel** [comment]
The documentation for installing PMD Plugin into JDeveloper is here: https://pmd.github.io/latest/pmd_userdocs_tools.html#jdeveloper

While there is no mention of JDeveloper 12c, it is in https://pmd.sourceforge.io/center.xml - but it also mentions `maxVersion="12.1.2.99"` - so, won't work with the latest version (12.2.1.4.0). (It would also only bring the very old PMD 5).

The source code for the plugin lives here: https://github.com/pmd/pmd-jdeveloper

Contributions for upgrading this to the latest JDeveloper version are welcome.
